### ISSUE TITLE
resolving/showing relative paths while typing them as arguments

### ISSUE THREAD

**Issue by lastfuture** [fish-shell/fish-shell#5343]
I have a suggestion regarding autocomplete/suggestions for relative paths going up the tree (e.g. `../../..`)

It would be awesome if as I typed a relative path I could see somewhere which full path this relative path would correspond to.

e.g. I'm in `/Users/username/git/16_0188_relaunch/wp-admin`, I start typing `cd ../..`, somewhere it would display to me that the full path I will end up in is `/Users/username/git`.

I'm envisioning the full path appearing below the prompt as a preview in the autosuggest color, but there is probably a way more in line with the look of the other completions.

What do you think?

**Comment by faho** [comment]
I don't think we want this as a default, especially since we don't really have a way of showing it.

What works is a binding that calls `realpath`:

```fish
bind \cg 'commandline -rt (builtin realpath -- (commandline -ct))'
```

will convert the current token to the full canonical path. You might want to check if it starts with `..` first, or you'll get $PWD prepended, but that's easy enough.

**Comment by floam** [comment]
I can think of two ways we might show this:

- as ["hints"](https://github.com/fish-shell/fish-shell/issues/2201), if we ever implement that.
- as completion descriptions in the pager

**Comment by faho** [comment]
>I think it should be marked "enhancement"

I'd roll it into #2201. If that doesn't get implemented, this won't either.

**Comment by lastfuture** [comment]
Thanks for considering it

**Comment by ridiculousfish** [comment]
Note #2201
